src/index.js

```diff
@@ -6,10 +6,9 @@ export const REDUX_OPERATION_SIGNATURE = '@@reduxOperations';
 
 export const walkState = (locationStack =[], state, initializer) => {
   return locationStack.reduce((reduction, key, currentIdx) => {
-    if (!reduction[key]) {
-      reduction[key] = (currentIdx === locationStack.length - 1) ? undefined : {};
-    }
-    return reduction[key] || initializer && initializer(undefined, {type:INIT_REDUX});
+    return reduction && reduction.hasOwnProperty(key) ? reduction[key] :
```

Comment:
the initial value of reduction is state, so we're guaranteed reduction is going to an object, meaning we can get rid of the first check, right?